driver/esp32/ili9XXX.py

```diff
@@ -576,3 +588,104 @@ def __init__(self,
             spihost, mhz, factor, hybrid, width, height, colormode, rot, invert, double_buffer, half_duplex, display_type=DISPLAY_TYPE_ILI9488,
             asynchronous=asynchronous, initialize=initialize)
 
+
+class gc9a01(ili9XXX):
+    # On the tested display the wirte direction and colormode appear to be
+    #reversed from how they are presented in the datasheet and so have been
+    #included here as new definitions
+
+    # GC9A01 displays handle rotation differently than ili9XXX displays
+    ROTATE = {
+        0: 0x40,
+        90: 0x20,
+        180: 0x80,
+        270: 0xE0
+    }
+
+    # GC9A01 displays handle colour mode differently than ili9XXX displays
+    COLOR_MODE = {
+        'RGB': 0x08,
+        'BGR': 0x00
+    }
+
```

Comment:
I understand that the meaning is reversed, but it's a bit confusing that for previous ili9XXX rot is set by `PORTRAIT`/`LANDSCAPE`, and for gc9a01 by number (180).  
I think it would be better to have the same notation on all controllers.  

What if gc9a01 recieved `PORTRAIT`/`LANDSCAPE`, the same way as ili9341, and translated it to the right values?
Same for `COLOR_MODE`, it's a bit confusing that colormode is a string on gc9a01 but a constant on the other controllers.

Reply:
I have no rechecked and it appears that portrait and landscape remain the same as the ili9341, I'm not sure why I thought it was different. I will submit a revised version.

However, is it worth allowing the option for any rotation (i.e. 0, 90, 180, 270)?

I have also added in an internal conversion for the RGB to BGR using the module defintions

Reply:
> However, is it worth allowing the option for any rotation (i.e. 0, 90, 180, 270)?

You can do that by adding additional constants next to `PORTRAIT` and `LANDSCAPE`.  
I'm not sure how to name them though...

Reply:
As far as I know, reverse portrait and reverse landscape are fairly standard terms.